libraries/ms-common/src/x86/gpio.c

```diff
@@ -1,9 +1,98 @@
 #include "gpio.h"
+#ifndef MPXE
```

Comment:
Instead of duplicating the entirety of x86/gpio.c, do you think it would be better to instead surround the MPXE-specific parts with `#ifdef MPXE`? It might look a little ugly but it might be better than duplicating all of the code.